packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,436 @@
+        vatID,
+        startPos,
+        endPos,
+      )) {
+        yield item;
+      }
+    }
+
+    if (startPos === endPos) {
+      return empty();
+    }
+
+    return reader();
+  }
+
+  const nl = Buffer.from('\n');
+
+  /**
+   * Read a transcript span and return it as a stream of data suitable for
+   * export to another store.  Items are terminated by newlines.
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be read
+   *
+   * @returns {AsyncIterable<Uint8Array>}
+   * @yields {Uint8Array}
+   */
+  async function* exportSpan(vatID, startPos) {
+    for (const entry of readSpan(vatID, startPos)) {
+      const item = typeof entry === 'string' ? Buffer.from(entry) : entry;
+      yield item;
+      yield nl;
+    }
+  }
+
+  const sqlAddItem = db.prepare(`
+    INSERT INTO transcriptItems (vatID, item, position)
+    VALUES (?, ?, ?)
+  `);
+
+  const sqlUpdateSpan = db.prepare(`
+    UPDATE transcriptSpans
+    SET endPos = ?, hash = ?, size = ?
+    WHERE vatID = ? AND current = 1
+  `);
+
+  /**
+   * Append an item to the current transcript span for a given vat
+   *
+   * @param {string} vatID  The whose transcript is being added to
+   * @param {string} item  The item to add
+   */
+  const addItem = (vatID, item) => {
+    ensureTxn();
+    const { startPos, endPos, size, hash } = getCurrentSpanBounds(vatID);
+    sqlAddItem.run(vatID, item, endPos);
+    const newEndPos = endPos + 1;
+    const newHash = computeItemHash(hash, item);
+    const newSize = size + item.length;
+    sqlUpdateSpan.run(newEndPos, newHash, newSize, vatID);
+    const rec = spanRec(vatID, startPos, newEndPos, newHash, newSize);
+    noteExport(currentSpanMetadataKey(rec), JSON.stringify(rec));
+  };
+
+  /**
+   * Import a transcript span from another store.
+   *
+   * @param {string} name  Artifact Name of the transcript span
+   * @param {SwingStoreExporter} exporter  Exporter from which to get the span data
+   * @param {object} info  Metadata describing the span
+   *
+   * @returns {Promise<void>}
+   */
+  async function importSpan(name, exporter, info) {
+    const parts = name.split('.');
+    const [type, vatID, rawStartPos, rawEndPos] = parts;
+    assert(
+      parts.length === 4 && type === 'transcript',
+      `expected artifact name of the form 'transcript.{vatID}.{startPos}.{endPos}', saw '${name}'`,
+    );
+    assert(
+      info.vatID === vatID,
+      `artifact name says vatID ${vatID}, metadata says ${info.vatID}`,
+    );
+    const startPos = Number(rawStartPos);
+    assert(
+      info.startPos === startPos,
+      `artifact name says startPos ${startPos}, metadata says ${info.startPos}`,
+    );
+    const endPos = Number(rawEndPos);
+    assert(
+      info.endPos === endPos,
+      `artifact name says endPos ${endPos}, metadata says ${info.endPos}`,
+    );
+    const artifactChunks = exporter.getArtifact(name);
```

Comment:
Same as for snapshots, this feels like the wrong layering.

```suggestion
  /**
   * Import a transcript span from another store.
   *
   * @param {AsyncIterable<Uint8Array>} artifactChunks
   * @param {{ vatID, startPos, endPos, hash, size }} info  Metadata describing the span
   *
   * @returns {Promise<void>}
   */
  async function importSpan(artifactChunks, info) {
```